### PR TITLE
Add Bubblesort method feature

### PR DESCRIPTION
**Pull request by carlos-osorio-developer**
# Add Bubblesort feature

> This feature will add a bubble_sort method to the project, which allows it to sort arrays in ascending order.

## To add

* bubble_sort.rb
* .rubocop.yml
* .gitignore
* .github/workflows/linters.yml
* .github/workflows/tests.yml

## To update
* README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,56 @@
-# bubble-sorting
-This is a Ruby code that allows you to sort an array using the bubble sorting method.
+![](https://img.shields.io/badge/Microverse-blueviolet)
+
+## Ruby Bubble-Sort and  Bubble-Sort_by project
+
+> This is a project made with ruby to order items inside of a array, one order items with numbers and the other one order strings inside of the array using a yield feature.
+
```
Comment (cyonii) [inline README.md:6]: Your README appear to have few defects, you can use [Grammarly](https://www.grammarly.com) to crosscheck and improve your grammar.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,56 @@
...
+### Setup
+
+Create a new folder/directory for the project
+
+### Install
+
+- open your terminal on your OS
+- go to the desired folder/directory that you want to have the project in
+- git clone gh repo clone https://github.com/carlos-osorio-developer/bubble-sorting/tree/bubblesort
```
Comment (cyonii) [inline README.md]: It appears you have made a typo here - `git clone gh repo clone ` -, please fix this typo.

**Review by cyonii — CHANGES_REQUESTED** [review]
# Status: Changes Required

Hello team, you have shown a great effort in this project, kudos to you.

## Highlights
- Good to have your linters configured and running green 🟢
- Excellent commit messages 👌
- Excellent job keeping your repo clean

## To Improve
You have a great project so far, but there are still vital issues you have to fix in order for this project to be approved, kindly take a look below.
- [x] Kindly update your PR description so that it describes the changes you intend to merge in. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [x] Please look at the inline comments and improve your README
- [x] Your `#bubble_sort_by` method failed to sort an array when block is passed, please this, it is vital for this project. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#assignment-1))
- [x] Please rename your `algorythm.rb` file to `bubble_sort.rb` as this name is important when running tests against your code.

>> ### You can find me on slack 👉 [CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) if you need further assistance or clarification. I'm always available and will be glad to help.

**Comment by carlos-osorio-developer** [comment]
Hello, thank you for your feedback. Regarding this comment: 

> * [x]  Your `#bubble_sort_by` method failed to sort an array when block is passed, please this, it is vital for this project. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#assignment-1))

That block is passed into the method at the end of the code, and it is working perfectly when running the command: 

```console
ruby bubble_sort.rb
```

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        sorted = false
+      end
+    end
+  end
+  array
+end
+
+puts "Sorted by array: #{bubble_sort_by(%w[hi hello hey]) { |left, right| left.length - right.length }}"
+
+puts "Sorted array: #{bubble_sort([5, 5, 6, 2, 1])}"
```
Comment (rukundoeric) [inline bubble_sort.rb]: debugging codes should be removed before you submit.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - [x] The project is clean. 💯
  - [x] The project follows Git flow. 💯

- ### To Improve
  - [x] [MANDATORY] Kindly remove all debugging codes.
  - [x] [MANDATORY] Your License and issue page  links are broken.
  
_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - [x] The project is clean. 💯
  - [x] The project follows Git flow. 💯

- ### To Improve 
  - [x] [MANDATORY] Your License and issue page links are returning `page not found`. kindly fix them before you request another code review.

_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by rukundoeric — APPROVED** [review]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - [x] The project is clean. 💯
  - [x] The project follows Git flow. 💯

- ### To Improve
  - N/A

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Pull request merged**